File: 2018.py
taget = int(input())
count = 1
start_index = 1
end_index = 1
sum = 1 


while end_index != taget :
  if sum == taget : #조건에 부합하는 경우
    count += 1
    end_index += 1
    sum += end_index

  elif sum > taget : # 부분합이 더 큰경우 
    sum -= start_index
    start_index += 1

  else : # 부분합이 더 적은경우
    end_index += 1
    sum += end_index


print(count)

File: 2018.c
#include <stdio.h>

int main(void) {
    int taget, cnt=1, l=1, r=1, sum= 1;
    scanf("%d", &taget);
    
    while (r != taget) {
        if (sum==taget) { // 부분합이 m과 같은 경우
            cnt++;r++;sum+=r;
            
        } else if (sum>taget) { // 부분합이 m보다 큰 경우
            sum-=l;l++;
            
        } else { // 부분합이 m보다 작은경우우
            r++;sum+=r;
            
        }
    }
    
    printf("%d", cnt);
    return 0;
}
